frontend/tests/unit/test_source_generator.py

```diff
@@ -110,8 +110,10 @@ def compile_and_return_output(directory, symbol, asm):
     objdump = sh.Command('objdump')
     output_file = f'{directory}/bin_output'
     gcc('-o', output_file, '-O0', glob.glob(f'{directory}/*.c'))
-    out = objdump(f'--disassemble={symbol}', output_file)
-    return bool(re.search(asm, str(out)))
+    out = objdump('-d', output_file)
```

Comment:
I think this function is actually testing whether a symbol exists in a compiled binary. Should we split this up into separate compile() and check_symbol_exists() functions?

Reply:
Good idea. Uploaded new patch set.